homeassistant/components/emoncms/sensor.py

```diff
@@ -84,19 +90,29 @@ async def async_setup_platform(
     exclude_feeds = config.get(CONF_EXCLUDE_FEEDID)
     include_only_feeds = config.get(CONF_ONLY_INCLUDE_FEEDID)
     sensor_names = config.get(CONF_SENSOR_NAMES)
+    scan_interval = config.get(CONF_SCAN_INTERVAL, timedelta(seconds=30))
 
     if value_template is not None:
         value_template.hass = hass
 
     emoncms_client = EmoncmsClient(url, apikey, session=async_get_clientsession(hass))
-    elems = await emoncms_client.async_list_feeds()
-
+    coordinator: DataUpdateCoordinator[list[dict[str, Any]] | None] = (
+        DataUpdateCoordinator(
+            hass,
+            _LOGGER,
+            name="emoncms",
+            update_method=emoncms_client.async_list_feeds,
+            update_interval=scan_interval,
+        )
+    )
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
This is not using a config flow, so this function won't work as well. I know @zxdavb is also looking into this so maybe he has a pointer for this

Reply:
I've removed the `async_config_entry_first_refresh()` to use directly the client during initialization

Just tell me if acceptable

Reply:
We should still use the coordinator to refresh, but I think you should `async_refresh` instead of the config entry one

Reply:
Using a **DataUpdateCoordinator** without Config Flow was quite straight-forward.

Assuming you have an awaitable `emoncms_client.async_list_feeds()`, your first step is as you have it:
```python
coordinator = DataUpdateCoordinator(
    hass,
    _LOGGER,
    name="emoncms",
    update_method=emoncms_client.async_list_feeds,
    update_interval=scan_interval,
)
```
... and to kick it off:
```python
await coordinator.async_config_entry_first_refresh() # get initial state
```

_HOWEVER, in between the two_, I had to add a fake listener, otherwise each refresh (including teh first, above) wouldn't schedule the next:
```python
# without a listener, DataUpdateCoordinator._schedule_refresh() won't be invoked by DataUpdateCoordinator._async_refresh()
coordinator.async_add_listener(lambda: None)
```

Reply:
But my point being is that `await coordinator.async_config_entry_first_refresh()` isn't available for non-config entry integrations. I thought you had to deal with this with evohome

Reply:
uses now `async_refresh` and not `async_config_entry_first_refresh`